tinyusb/src/class/hid/hid_device.h

```diff
@@ -265,21 +265,23 @@ TU_ATTR_WEAK bool tud_hid_set_idle_cb(uint8_t idle_rate);
     HID_REPORT_COUNT ( 16                                     ) ,\
     HID_REPORT_SIZE  ( 1                                      ) ,\
     HID_INPUT        ( HID_DATA | HID_VARIABLE | HID_ABSOLUTE ) ,\
-    /* 8 bit X, Y, Z, Rz (min -127, max 127 ) */ \
+    /* 8 bit X, Y, Z, Rx, Ry, Rz (min -127, max 127 ) */ \
     HID_USAGE_PAGE   ( HID_USAGE_PAGE_DESKTOP                 ) ,\
     HID_USAGE        ( HID_USAGE_DESKTOP_X                    ) ,\
     HID_USAGE        ( HID_USAGE_DESKTOP_Y                    ) ,\
     HID_USAGE        ( HID_USAGE_DESKTOP_Z                    ) ,\
+    HID_USAGE        ( HID_USAGE_DESKTOP_RX                   ) ,\
+    HID_USAGE        ( HID_USAGE_DESKTOP_RY                   ) ,\
     HID_USAGE        ( HID_USAGE_DESKTOP_RZ                   ) ,\
     HID_LOGICAL_MIN  ( 0x81                                   ) ,\
     HID_LOGICAL_MAX  ( 0x7f                                   ) ,\
-    HID_REPORT_COUNT ( 4                                      ) ,\
+    HID_REPORT_COUNT ( 6                                      ) ,\
     HID_REPORT_SIZE  ( 8                                      ) ,\
     HID_INPUT        ( HID_DATA | HID_VARIABLE | HID_ABSOLUTE ) ,\
     /* 8 bit Hat Button Map  */ \
     HID_USAGE_PAGE   ( HID_USAGE_PAGE_DESKTOP                 ) ,\
     HID_USAGE        ( HID_USAGE_DESKTOP_HAT_SWITCH           ) ,\
-    HID_LOGICAL_MIN  ( 0                                      ) ,\
+    HID_LOGICAL_MIN  ( 1                                      ) ,\

```

Comment:
I had to restore this value to 1 as I get a weird behaviour. After a little research I realized there is no a "standard" gamepad hid report, each manufacturer implements its own. For example Sony maps his X, Y, Z, RX, RY, RY from 0 to 255 meanwhile Windows/Linux maps those descriptors from -127 to 127. Aswell the DPAD is diffently mapped on the PS4 controller because the '0' value is mapped as 'UP' (https://github.com/felis/USB_Host_Shield_2.0/blob/master/PS4Parser.cpp). That's the reason the descriptor is mapped from 0 to 7 instead of 1 to 8.

Reply:
thanks for the explanation. That makes sense to me. I am ok with `1 to 8` as long as it works with most OS.